### PR TITLE
Revert "[SPARK-34415][ML] Randomization in hyperparameter optimization"

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Revert https://github.com/apache/spark/commit/397b843890db974a0534394b1907d33d62c2b888 and https://github.com/apache/spark/commit/5a48eb8d00faee3a7c8f023c0699296e22edb893 

### Why are the changes needed?

As discussed in https://github.com/apache/spark/pull/33800#issuecomment-904140869, there is correctness issue in the current implementation. Let's revert the code changes from branch 3.2 and fix it on master branch later

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Ci tests

### CODE REVIEW

**Comment by gengliangwang** [comment]
cc @PhillHenry as well.

**Review by srowen — APPROVED** [review]
Yeah I think this is best for now. LGTM

**Comment by srowen** [comment]
Oh, we need to revert from master too BTW

**Comment by gengliangwang** [comment]
> Oh, we need to revert from master too BTW

@srowen ok I thought we are going to fix it on master recently.  Anyway, I have rebased the PR against the master branch.

**Comment by srowen** [comment]
Yeah I think it's best to revert entirely for now - the follow-up change will be different enough that it probably just needs a new start

File: python/pyspark/ml/tests/test_tuning.py [hunk 1]
```diff
@@ -16,10 +16,8 @@
 #
 
 import tempfile
-import math
 import unittest
 
-import numpy as np
```
Comment (srowen) [inline python/pyspark/ml/tests/test_tuning.py:22]: Ah, sorry for the trouble, but a later commit now needs this numpy import:

```
flake8 checks failed:
./python/pyspark/ml/tests/test_tuning.py:100:16: F405 'np' may be undefined, or defined from star imports: pyspark.ml.tests.test_tuning
        assert np.allclose(avg_metrics, [2.33333333, 2.5, 2.03333333, 5.33333333])
               ^
./python/pyspark/ml/tests/test_tuning.py:101:16: F405 'np' may be undefined, or defined from star imports: pyspark.ml.tests.test_tuning
        assert np.allclose(std_metrics, [0.94280904, 0.40824829, 0.04714045, 1.88561808])
               ^
./python/pyspark/ml/tests/test_tuning.py:143:25: F405 'np' may be undefined, or defined from star imports: pyspark.ml.tests.test_tuning
        self.assertTrue(np.allclose(cvModel.stdMetrics, cvModelCopied.stdMetrics))
                        ^
3     F405 'np' may be undefined, or defined from star imports: pyspark.ml.tests.test_tuning
```

Just don't revert this line

**Review by srowen — CHANGES_REQUESTED** [review]
Also we need to revert this follow up commit:

https://github.com/apache/spark/pull/31687/files

Sorry for the trouble on this one, but that should be it.

File: python/pyspark/ml/tests/test_tuning.py (hunk 1 above)
Comment (gengliangwang) [reply python/pyspark/ml/tests/test_tuning.py:22]: Thanks a lot!

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Pull request closed without merging**